### ISSUE TITLE
Lint/Syntax problems with rubocop 0.53.0 & ruby 2.3.2

### ISSUE THREAD

**Issue by bertocq** [bbatsov/rubocop#5639]
While [updating rubocop to 0.53.0 at consul](https://github.com/consul/consul/compare/rubocop_update?expand=1) I found out this problem by running rubocop over the whole codebase.

At [`/app/models/abilities/common.rb#L22-L24`](https://github.com/consul/consul/blob/1471c97af8d66c260a5c2a966154cfd675aa643f/app/models/abilities/common.rb#L22-L24)

We've this code:
```
cannot [:edit, :update], Legislation::Proposal do |proposal|
  proposal.editable_by?(user)
end
```
And two Lint/Syntax offenses appear because of the first line, that we didn't had with rubocop 0.52.1 (previous release). 

The curious part is that if I change the symbol array to the %i format (`%i[edit update]`) the offenses dissapear. Also that this doesn't happen with at least ruby 2.5.0 and 2.4.1

Thanks! 😃 

--------

## Expected behavior

Rubocop should not find any offenses

## Actual behavior

Rubocop returns:
```ruby
Inspecting 1 file
E

Offenses:

example.rb:9:52: E: Lint/Syntax: unexpected token kDO
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
    cannot [:edit, :update], Legislation::Proposal do |proposal|
                                                   ^^
example.rb:13:1: E: Lint/Syntax: unexpected token kEND
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
end
^^^

1 file inspected, 2 offenses detected
```
## Steps to reproduce the problem

1. Create a `example.rb` file with:
```ruby
# frozen_string_literal: true

class Common
  def initialize(user)
     cannot [:edit, :update], Legislation::Proposal do |proposal|
      proposal.editable_by?(user)
    end

	cannot %i[edit update], Legislation::Proposal do |proposal|
      proposal.editable_by?(user)
    end
  end
end
```
2. Install ruby 2.3.2 and use it. Also install rubocop 0.53.0
3. Run `rubocop example.rb`
4. You'll see the first `cannot` will raise an offense, but the second will not.

## RuboCop version
```
$ rubocop -V
0.53.0 (using Parser 2.5.0.2, running on ruby 2.3.2 x86_64-darwin16)
```

**Comment by pocke** [comment]
It is parser gem problem. And probably it has been fixed in parser 2.5.0.3 (You use 2.5.0.2).
Can you try updating parser gem?

**Comment by bertocq** [comment]
Yes you are right! I did `gem update parser` and `rubocop -V` output was:
```bash
0.53.0 (using Parser 2.5.0.3, running on ruby 2.3.2 x86_64-darwin16)
```
Finally `rubocop example.rb` came out all green 😊 

Sorry for this issue being a duplicate of https://github.com/bbatsov/rubocop/issues/5573 🙇 I tried to search my error in existent issues before opening but clearly my search wasn't good enough :( I found it now trying to figure out how you knew it was a parser issue (yesterday I wasn't able so I opened the issue)

Thanks!
